### PR TITLE
Add Artifact Synchronization docs

### PR DESCRIPTION
**Pull request by shaniR**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Review by tharindu1st — APPROVED** [review]

File: en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+    ```
+    [apim.sync_runtime_artifacts.publisher]
+    artifact_saver = "DBSaver"
+    publish_directly_to_gateway = "true"
+    ```
+    
+    | Configuration                                    | Description                          |
+    | -------------------------------------------------| ------------------------------------ |
+    |apim.sync_runtime_artifacts.publisher            | configuration element then all the artifacts will be saved to the storage via  the extension point. In default they will get stored in database|
+    |artifact_saver|specify the extension point. The default is `DBSaver` where the artifacts are saved in the database.|
```
Comment (isharac) [inline en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md:67]: what are the possible values for this?

File: en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    artifact_retriever = "DBRetriever"
+    deployment_retry_duartion = 15000
+    data_retrieval_mode = "sync"
+    save_artifacts_locally = false
+    ```
+ 
+    |Configuration| Description|
+    |--------------------------------------------------|------------------------------------------------|
+    |gateway_labels|specify the labels which the gateway is going to subscribe to. Only the APIs with these labels will be pulled from the extension point and deployed.|
+    |artifact_retriever|  we can specify the extension point. The default is `DBRetriever` where the artifacts are pulled from the database.|
```
Comment (isharac) [inline en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md:91]: Any other possible values?

File: en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+     |save_artifacts_locally|If we add the config `save_artifacts_locally = true` or remove `save_artifacts_locally` config  then synapseartifacts will be stored in the file system. (Saved in `<APIM_HOME>/repository/deployment/server/synapse-configs/default/`) . When `save_artifacts_locally = false` then the artifacts from extension point will not be stored in the file system.|
+    
+ 3. If the gateway profile is running with an port offset add the following configuration.
+     ```
+     [apim.event_hub]
```
Comment (isharac) [inline en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md:98]: we can add this as tab="Format" , tab="Example"

File: en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+## Artifact Synchronization with a Shared File System
+
+To enable synchronization for runtime artifacts of the two all in one WSO2 API-M instances, it is recommended to have a
+shared file system. Configure a shared file system as the content synchronization mechanism. You can use a common shared file 
+system such as Network File System (NFS) or any other shared file system that is available. 
+
+You need to mount the `<API-M_HOME>/repository/deployment/server` directory of the two nodes to the shared file system, 
+in order to share all APIs and throttling policies between all the nodes.
+
```
Comment (isharac) [inline en/docs/install-and-setup/setup/distributed-deployment/synchronizing-artifacts-in-a-gateway-cluster.md:17]: If multi-tenancy is used we need to mount <API-M_HOME>/repository/tenants

**Pull request merged**